presto-orc/src/test/java/com/facebook/presto/orc/TestOrcBloomFilters.java

```diff
@@ -0,0 +1,333 @@
+
+        // Re-construct
+        HiveBloomFilter hiveBloomFilter = new HiveBloomFilter(ImmutableList.copyOf(Longs.asList(bloomFilter.getBitSet())), bloomFilter.getBitSize(), bloomFilter.getNumHashFunctions());
+
+        // String
+        assertTrue(hiveBloomFilter.testString(TEST_STRING));
+        assertFalse(hiveBloomFilter.testString(TEST_STRING_NOT_WRITTEN));
+
+        // Integer
+        assertTrue(hiveBloomFilter.testLong(TEST_INTEGER));
+        assertFalse(hiveBloomFilter.testLong(TEST_INTEGER + 1));
+    }
+
+    public void compareBitset(long[] a, long[] b)
+    {
+        int i = 0;
+        for (long l : a) {
+            assertEquals(l, b[i]);
+            i++;
+        }
+    }
+
+    public byte[] serializeBloomFilterToIndex(OrcProto.BloomFilter bloomFilter, OrcProto.BloomFilterIndex bloomFilterIndex)
+            throws IOException
+    {
+        assertTrue(bloomFilter.isInitialized());
+
+        OrcProto.BloomFilterIndex.Builder builder = bloomFilterIndex.toBuilder();
+        builder.addBloomFilter(bloomFilter);
+
+        OrcProto.BloomFilterIndex index = builder.build();
+        assertTrue(index.isInitialized());
+        assertEquals(index.getBloomFilterCount(), 1);
+
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+        index.writeTo(os);
+        os.flush();
+        return os.toByteArray();
+    }
+
+    @Test
+    public void testOrcHiveBloomFilterSerde()
+            throws Exception
+    {
+        BloomFilter bloomFilterWrite = new BloomFilter(1000L, 0.05D);
+
+        bloomFilterWrite.addString(TEST_STRING);
+        assertTrue(bloomFilterWrite.testString(TEST_STRING));
+
+        OrcProto.BloomFilter.Builder bloomFilterBuilder = OrcProto.BloomFilter.newBuilder();
+        for (long l : bloomFilterWrite.getBitSet()) {
+            bloomFilterBuilder.addBitset(l);
+        }
+        bloomFilterBuilder.setNumHashFunctions(bloomFilterWrite.getNumHashFunctions());
+
+        OrcProto.BloomFilter bloomFilter = bloomFilterBuilder.build();
+        OrcProto.BloomFilterIndex bloomFilterIndex = OrcProto.BloomFilterIndex.getDefaultInstance();
+        byte[] bytes = serializeBloomFilterToIndex(bloomFilter, bloomFilterIndex);
+
+        // Read through method
+        InputStream inputStream = new ByteArrayInputStream(bytes);
+        OrcMetadataReader metadataReader = new OrcMetadataReader();
+        List<HiveBloomFilter> bloomFilters = metadataReader.readBloomFilterIndexes(inputStream);
+
+        assertEquals(bloomFilters.size(), 1);
+
+        assertTrue(bloomFilters.get(0).testString(TEST_STRING));
+        assertFalse(bloomFilters.get(0).testString(TEST_STRING_NOT_WRITTEN));
+
+        assertEquals(bloomFilterWrite.getBitSize(), bloomFilters.get(0).getBitSize());
+        assertEquals(bloomFilterWrite.getNumHashFunctions(), bloomFilters.get(0).getNumHashFunctions());
+
+        // Validate bitset
+        compareBitset(bloomFilters.get(0).getBitSet(), bloomFilterWrite.getBitSet());
```

Comment:
Just use `assertTrue(Arrays.equals(bloomFilters.get(0).getBitSet(), bloomFilterWrite.getBitSet()));` and remove the `compareBitset` method